From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Mullin <tim@cpanel.net>
Date: Thu, 19 Jan 2023 15:39:18 +0000
Subject: [PATCH 22/22] mod_http2: client resets of HTTP/2 streams led to
 unwanted 500 errors reported in access logs and error documents. The
 processing of the reset was correct, only unneccesary reporting was caused.

From:
https://svn.apache.org/viewvc?view=revision&revision=1906775
---
 modules/http2/h2_c2_filter.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/http2/h2_c2_filter.c b/modules/http2/h2_c2_filter.c
index f537a19..37254fc 100644
--- a/modules/http2/h2_c2_filter.c
+++ b/modules/http2/h2_c2_filter.c
@@ -615,7 +615,7 @@ apr_status_t h2_c2_filter_catch_h1_out(ap_filter_t* f, apr_bucket_brigade* bb)
     ap_assert(conn_ctx);
     H2_FILTER_LOG("c2_catch_h1_out", f->c, APLOG_TRACE2, 0, "check", bb);
 
-    if (!conn_ctx->has_final_response) {
+    if (!f->c->aborted && !conn_ctx->has_final_response) {
         if (!parser) {
             parser = apr_pcalloc(f->c->pool, sizeof(*parser));
             parser->id = apr_psprintf(f->c->pool, "%s-%d", conn_ctx->id, conn_ctx->stream_id);
